machine.go

```diff
@@ -707,6 +707,13 @@ func (m *Machine) UpdateGuestDrive(ctx context.Context, driveID, pathOnHost stri
 	return nil
 }
 
+// IsAlive makes sure that a Firecracker instance is running and responds via API by
+// calling `DescribeInstance` endpoint behind.
+func (m *Machine) IsAlive(ctx context.Context) error {
```

Comment:
It looks like one of the valid states that can be returned by `DescribeInstance` is `Halted`: https://github.com/firecracker-microvm/firecracker/blob/35b73437641122865b1603bd74c144035e1790c9/api_server/swagger/firecracker.yaml#L453

There's also `Uninitialized`. That makes me feel like `IsAlive` is not really what one would expect. It seems more like it's checking whether the VM manager is alive rather than what we'd consider the `Machine`.

If we want to keep the behavior as is, I can't think of any great names, but maybe something like `Exists` would be better than `IsAlive`?

On the other hand, would it make sense to change the behavior and return an error here if the State is `Halted` or `Uninitialized`? What's the actual motivation behind this change?

Reply:
(I accidentally clicked the approve button, I think this should be addressed before merging)

Reply:
(@sipsma - I've "dismissed" your review per your comment here.)

Reply:
Some suggestions for names: `IsVMMAlive`, `IsVMMResponsive`, `Reachable`